pkg/storage/tsdb/config.go

```diff
@@ -64,6 +65,10 @@ type BucketConfig struct {
 	GCS        gcs.Config        `yaml:"gcs"`
 	Azure      azure.Config      `yaml:"azure"`
 	Filesystem filesystem.Config `yaml:"filesystem"`
+
+	// Not used internally, meant to allow callers to wrap Buckets
+	// created using this config
+	Middlewares []func(objstore.Bucket) objstore.Bucket `yaml:"-"`
```

Comment:
Might be good to have an option to return error too.

Reply:
👍